lib/interjectable/rspec.rb

```diff
@@ -107,21 +114,27 @@ def test_inject(dependency, &setter)
       # for the same #test_inject call since those before hooks only run once,
       # and therefore only setup a single after hook.
       return if scope == :each && RESTORE_HOOKS[key].any? { |group| rspec_example_group <= group }
-      # if dependency == :dependency && scope == :each
-      #   puts "adding new after=#{key.inspect} hooks=#{RESTORE_HOOKS[key]} group=#{rspec_example_group}"
-      # end
       RESTORE_HOOKS[key] << rspec_example_group
 
-      # if dependency == :dependency && scope == :each
-      #   puts RESTORE_HOOKS.select { |(_, d, s)| d == :dependency && s == :each }
-      # end
-
       rspec_example_group.after(scope) do
-        # if dependency == :dependency && scope == :each
-        #   puts "restore:  #{key.inspect} #{rspec_example_group}"
-        # end
         injector.restore
       end
     end
   end
+
+  module RSpecHelper
+    def test_inject(target, dependency, value)
+      unless value
+        raise ArgumentError, "missing value for #{dependency.inspect}, correct usage: test_inject(my_thing, #{dependency.inspect}, FakeDependency.new)"
+      end
+
+      ClassMethods.test_inject(self.class, target, dependency, value)
+    end
+  end
+end
+
+if defined?(RSpec)
+  RSpec.configure do |c|
+    c.include(Interjectable::RSpecHelper)
+  end
```

Comment:
```suggestion
  end
else
  warn "RSpec helper was required but RSpec has not beed defined"
```

What if we added a warning for this case, like this?

Reply:
I like it. Adding. `warn` only or should I abort?

Reply:
I don't know what's expected of a helper like this....I agree that it seems like an error, so let's error? Or maybe throw? I trust you on this.

Reply:
I think throwing is better in this case.